torch/utils/data/datapipes/iter/combining.py

```diff
@@ -338,9 +379,6 @@ def get_next_element_by_instance(self, instance_id: int):
                     self.main_datapipe_exhausted = True
                     self._datapipe_iterator = None
 
-    def is_instance_started(self, instance_id: int) -> bool:
-        return self.instance_started[instance_id]
-
     def is_every_instance_exhausted(self) -> bool:
         return self.main_datapipe_exhausted and all(not child_buffer for child_buffer in self.child_buffers)
```

Comment:
Is this still correct? Why are we checking if the input datapipe is exhausted? This leads to being warned in very simple cases where this shouldn't be happening:

```py
from torchdata.datapipes.iter import IterableWrapper, Demultiplexer, Zipper

dp = IterableWrapper(range(6))
dps = Demultiplexer(dp, 3, lambda x: x % 3)
dp = Zipper(*dps)

next(iter(dp))
```

Reply:
Thanks for pointing that out. I suspected something is off. 

The method is still correct but needs to be moved to a different place (right before `reset()` is called).